impl-endpoints/cloudflare-go/runner.go

```diff
@@ -0,0 +1,76 @@
+package main
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"flag"
+	"io/ioutil"
+	"log"
+	"os"
+)
+
+type testCase interface {
+	ClientConfig() (*tls.Config, error)
+	ClientConnectionHandler(*tls.Conn) bool
+	// ServerConfig() *tls.Config
+	// ServerListenerHandler(net.Listener) bool
+}
+
+type delegatedCredentialTestCase struct {
+}
+
+func (c delegatedCredentialTestCase) ClientConfig() (*tls.Config, error) {
+	certPool := x509.NewCertPool()
+	pem, err := ioutil.ReadFile("/certs/rootCA.pem")
+	if err != nil {
+		return nil, err
+	}
+
+	certPool.AppendCertsFromPEM(pem)
+	return &tls.Config{
+		RootCAs:                    certPool,
+		SupportDelegatedCredential: true,
+	}, nil
+}
+
+func (c delegatedCredentialTestCase) ClientConnectionHandler(conn *tls.Conn) bool {
+	state := conn.ConnectionState()
+	return state.VerifiedDC != nil
+}
+
+var testCaseHandlers = map[string]testCase{
+	"dc": delegatedCredentialTestCase{},
+}
+
+func main() {
+	testCase := flag.String("test", "", "Test case identifier")
+	role := flag.String("role", "client", "Role string (client or server)")
+	flag.Parse()
+
+	handler, ok := testCaseHandlers[*testCase]
+	if !ok {
+		// Skip this test case
+		os.Exit(-2)
```

Comment:
@xvzcf this return code was arbitrary -- we can change it to whatever makes sense!